libraries/Matter/src/MatterEndpoints/MatterTemperatureSensor.h

```diff
@@ -0,0 +1,63 @@
+// Copyright 2024 Espressif Systems (Shanghai) PTE LTD
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+#include <sdkconfig.h>
+#ifdef CONFIG_ESP_MATTER_ENABLE_DATA_MODEL
+
+#include <Matter.h>
+#include <MatterEndPoint.h>
+
+class MatterTemperatureSensor : public MatterEndPoint {
+public:
+  MatterTemperatureSensor();
+  ~MatterTemperatureSensor();
+  // default initial raw temperature
+  virtual bool begin(int16_t _rawTemperature = 0);
+  bool begin(double temperatureCelcius) {
+    int16_t rawValue = static_cast<int16_t>(temperatureCelcius * 100.0f);
+    return begin(rawValue);
+  }
+  // this will just stop processing Temperature Sensor Matter events
+  void end();
+
+  // set the reported raw temperature
+  bool setRawTemperature(int16_t _rawTemperature);
```

Comment:
Since "raw" here just means tC*100, I will suggest to move that method to private. You could then add two new methods for Fahrenheit.